X16 Reference - 11 -  Working with CMDR-DOS.md

```diff
@@ -0,0 +1,368 @@
+
+To open the command channel, you can use the OPEN command with secondary address 15.
+
+`10 OPEN 15,8,15`
+
+If you want to issue a command immediately, add your command string at the end of the OPEN statement: 
+
+`10 OPEN 15,8,15, "CD:/"`
+
+This example changes to the root directory of your SD card. 
+
+Now you can check your status by reading ST: 
+
+`20 IF ST=64 THEN PRINT "OK": GOTO 50`
+
+To actually read the error channel and clear the error status, you need to read four values: 
+
+`30 INPUT#15,A,B$,C,D`
+
+A is the error number. B$ is the error message. C and D are unused in CBM-DOS, but will return the track and sector when
+used with a Commodore disk drive. 
+
+```
+40 PRINT A;B$;C;D
+50 CLOSE 15
+```
+
+So the entire program looks like: 
+
+```BASIC
+10 OPEN 15,8,15, "CD:/"
+20 IF ST=64 THEN PRINT "OK": GOTO 50
+30 INPUT#15,A,B$,C,D
+40 PRINT A;B$;C;D
+50 CLOSE 15
+```
+
+You can also use the DOS command to send a command to CMDR-DOS. Entering DOS by itself will print the drive's status on
+the screen. Entering a command in quotes or a string variable will execute the command. We will talk more about the 
+status variable and DOS status message in the next section. 
+
+```
+DOS
+00, 0K, 00, 00
+READY.
+DOS "CD:/"
+```
+
+The special case of `DOS "$"` will print a directory listing.
+
+`DOS "$"`
+
+You can also read the name of the current directory with "$=C"
+
+`DOS "$=C"`
+
+## DOS Features
+
+This is the base features set compared to other Commodore DOS devices:
+
+| Feature          | 1541 | 1571/1581 | CMD HD/FD | SD2IEC   | *CMDR-DOS*      |
+|------------------|------|-----------|-----------|----------|-----------------|
+| Sequential files | yes  | yes       | yes       | yes      | yes             |
+| Relative files   | yes  | yes       | yes       | yes      | not yet         |
+| Block access     | yes  | yes       | yes       | yes      | not yet         |
+| Code execution   | yes  | yes       | yes       | no       | yes             |
+| Burst commands   | no   | yes       | yes       | no       | no              |
+| Timestamps       | no   | no        | yes       | yes      | yes             |
+| Time API         | no   | no        | yes       | yes      | not yet         |
+| Partitions       | no   | no        | yes       | yes      | yes             |
+| Subdirectories   | no   | no        | yes       | yes      | yes             |
+
+It consists of the following components:
+
+* Commodore DOS interface
+	* `main.s`: TALK/LISTEN dispatching
+	* `parser.s`: filename/path parsing
+	* `cmdch.s`: command channel parsing, status messages
+	* `file.s`: file read/write
+* FAT32 interface
+	* `match.s`: FAT32 character set conversion, wildcard matching
+	* `dir.s`: FAT32 directory listing
+	* `function.s`: command implementations for FAT32
+* FAT32 implementation
+	* `fat32/*`: [FAT32 for 65c02 library](https://github.com/commanderx16/x16-rom/tree/master/dos/fat32)
```

Comment:
Since we're on the community side of docs, perhaps we should link to our fork's version.

Reply:
done